src/main/java/at/hannibal2/skyhanni/data/mob/Mob.kt

```diff
@@ -166,42 +174,44 @@ class Mob(
         fullEntityList().associateWith { this }
 
     internal fun internalAddEntity(entity: EntityLivingBase) {
+        internalRemoveColor()
         if (baseEntity.entityId > entity.entityId) {
             extraEntitiesList.add(0, baseEntity)
-            internalRemoveColor()
             baseEntity = entity
-            internalHighlight()
         } else {
             extraEntitiesList.add(extraEntitiesList.lastIndex + 1, entity)
         }
+        internalHighlight()
         updateBoundingBox()
         MobData.entityToMob[entity] = this
     }
 
     internal fun internalAddEntity(entities: Collection<EntityLivingBase>) {
         val list = entities.drop(1).toMutableList().apply { add(baseEntity) }
-        extraEntitiesList.addAll(0, list)
         internalRemoveColor()
+        extraEntitiesList.addAll(0, list)
         baseEntity = entities.first()
         internalHighlight()
         updateBoundingBox()
         removeExtraEntitiesFromChecking()
         MobData.entityToMob.putAll(entities.associateWith { this })
     }
 
-    internal fun internalUpdateOfEntity(entity: EntityLivingBase) = when (entity.entityId) {
-        baseEntity.entityId -> {
-            internalRemoveColor()
-            baseEntity = entity
-            internalHighlight()
-        }
-
-        armorStand?.entityId ?: Int.MIN_VALUE -> armorStand = entity as EntityArmorStand
-        else -> {
-            extraEntitiesList.remove(entity)
-            extraEntitiesList.add(entity)
-            Unit // To make return type of this branch Unit
+    internal fun internalUpdateOfEntity(entity: EntityLivingBase) {
+        internalRemoveColor()
+        when (entity.entityId) {
+            baseEntity.entityId -> {
+                baseEntity = entity
+            }
+
+            armorStand?.entityId ?: Int.MIN_VALUE -> armorStand = entity as EntityArmorStand
```

Comment:
instead of `?:` you can use `, `. 
also, for the case that armor stand is not null, isnt this the same as with base entity? why should both have the same id but be different entity objects?
what is the case where entity.entityId is min value so that the min value check can be true?